### DISCUSSION TITLE
System.Drawing.Common is going windows only

### DISCUSSION THREAD

**Discussion by trampster** [xamarin/Xamarin.Forms#14451] [Q&A]
Announcement here: https://github.com/dotnet/designs/blob/main/accepted/2021/system-drawing-win-only/system-drawing-win-only.md

What effect will this have on Xamarin projects?

With .net Maui and Xamarin moving to use .Net 6 libraries, will we therefore lose support for System.Drawing.* namespaces?

Currently I'm using System.Drawing.Color and System.Drawing.Image, do I need to migrate away from using these?

It also seems that Xamarin.Forms itself uses the System.Drawing.* namespace.

**Comment by mcguu** [comment]
can you use system.drawinng in xamarin.forms?

**Reply by trampster** [reply]
Yes we use System.Drawing.Color and System.Drawing.Image in our Xamarin.Forms app. 

The Xamarin.Forms code itself has 78 references to System.Drawing including in it's iOS, Mac and Linux implementations.

**Reply by mcguu** [reply]
But you have to add `System.Drawing.Color` and `System.Drawing.Image` in nuget right?

just in the xf template there is no System.Drawings.

I think If you can add it to maui then you can use it too.

**Reply by trampster** [reply]
I don't think it requires any additional nuget package currently.

Regarding .NET Maui it will use .Net 6 and in .Net 6 System.Drawing.Common will be windows only, So you will not be able to add a nuget reference to it.

**Reply by mcguu** [reply]
In my project I can not use it directly.

If there is no System.Drawning you have to write the function yourself right?

you can write a interface and write the implementaion in android ios mac and windows it is complicated

**Reply by trampster** [reply]
Xamarin.Essentials has method for converting to and from System.Drawing.Color, So maybe it is provided by Xamarin.Essentials.
